app/src/main/java/com/esri/arcgisruntime/opensourceapps/datacollection/views/popup/PopupView.kt

```diff
@@ -168,33 +180,152 @@ class PopupView : FrameLayout {
+                                popupField,
+                                popupFieldEditValue.text.toString()
+                            )
+                            if (validationError != null) {
+                                val fieldLabelWithValidationError =
+                                    popupField.label + ": " + validationError.message
+                                popupFieldLabel.text = fieldLabelWithValidationError
+                                popupFieldLabel.setTextColor(Color.RED)
+                            } else {
+                                popupFieldLabel.text = popupField.label
+                                popupFieldLabel.setTextColor(oldColors)
+                            }
                         }
                     }
                 }
             } else {
                 popupFieldEditValue.visibility = View.GONE
+                popupFieldCvdValues.visibility = View.GONE
                 popupFieldValue.visibility = View.VISIBLE
             }
         }
+
+        /**
+         * Sets up spinner for PopupFields that have a CodedValueDomain.
+         */
+        private fun setUpSpinner(codedValueDomain: CodedValueDomain, popupField: PopupField) {
+
+            val codedValuesNames = mutableListOf<String>()
+            for (codedValue in codedValueDomain.codedValues) {
+                codedValuesNames.add(codedValue.name)
+            }
+            popupFieldCvdValues.adapter = ArrayAdapter(
+                binding.root.context,
+                android.R.layout.simple_spinner_dropdown_item,
+                codedValuesNames
+            )
+            val codedValuePosition = getCodedValuePosition(
+                popupManager.getFieldValue(popupField),
+                codedValueDomain.codedValues
+            )
+            // set the PopupField value as selected in the spinner
+            popupFieldCvdValues.setSelection(codedValuePosition)
+            popupFieldCvdValues.onItemSelectedListener =
+                object : AdapterView.OnItemSelectedListener {
+                    override fun onNothingSelected(parent: AdapterView<*>?) {
+                        TODO("Not yet implemented")
+                    }
+
+                    override fun onItemSelected(
+                        parent: AdapterView<*>?,
+                        view: View?,
+                        position: Int,
+                        id: Long
+                    ) {
+                        popupManager.updateValue(
+                            codedValueDomain.codedValues[position].code,
+                            popupField
+                        )
+                    }
+                }
+        }
+
+        /**
+         * Updates the value of the specified PopupField to the appropriately cast string value of
+         * the specified value
+         */
+        private fun updateValue(popupField: PopupField, newValue: String): ArcGISRuntimeException? {
```

Comment:
Add UpdateValue function to call PopupManager.updateValue() with the newValue of String type cast to the appropriate type based on the Popup Fields type. (If we don't do this PopupManager.updateValue() throws exception stating the value is not the expected type)